include/camp/defines.in.hpp

```diff
@@ -13,25 +13,31 @@ For details about use and distribution, please read LICENSE and NOTICE from
 
 #include <cstddef>
 #include <cstdint>
-#include <string>
 #include <stdexcept>
+#include <string>
+
+// Define CAMP_CONFIG_OVERRIDE to change this on a per-file basis
+#if !defined(CAMP_CONFIG_OVERRIDE)
+#cmakedefine CAMP_ENABLE_OPENMP
+#cmakedefine CAMP_ENABLE_TARGET_OPENMP
+#cmakedefine CAMP_ENABLE_CUDA
+#cmakedefine CAMP_ENABLE_HIP
+#cmakedefine CAMP_ENABLE_SYCL
+#endif
 
-#if defined(__CUDACC__) || defined(CAMP_HAVE_CUDA)
+// include cuda header if configured, even if not in use
```

Comment:
Is this necessary? I thought we were trying to allow people to use non-GPU compilers with GPU enabled headers (https://github.com/LLNL/RAJA/pull/1114).

Reply:
That's correct.  This actually supports that better than before.  It pulls in the compiler-agnostic header if it's enabled during configure, but doesn't require that the file be compiled with a cuda or hip compiler.

Reply:
Ok, does the documentation note that if you use a non-GPU compiler that you'll have to point it to the GPU compiler's header? Is there any way to detect if the header is not available and give a useful error?

Reply:
Since camp now uses BLT, if rocm or cuda can't be found it should error out there when it tries to resolve the *_runtime dependency.  Is that what you had in mind?

Reply:
I was thinking of __has_include, but that's c++17. I'm trying to keep non-cmake projects from shooting themselves in the foot.

Reply:
The only way I know of to work around that and provide an error would require us to provide a header in a directory that gets searched after all others that contains a header of the same name to detect this.  If we're talking about projects that don't use cmake, we could do it for those that use pkg_config, but beyond that we can't detect it.

Reply:
Ok, its probably not worth it then. I just foresee people being confused who aren't aware you can use the cuda runtime without using cuda.